src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/EnvironmentUrlHardcodedRuleTests.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Rules;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Linq;
+
+// TODO: Test with different configs
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
+{
+    [TestClass]
+    public class EnvironmentUrlHardcodedRuleTests : LinterRuleTestsBase
+    {
+        private void CompileAndTest(string text, int expectedDiagnosticCount)
+        {
+            using (new AssertionScope($"linter errors for this code:\n{text}\n"))
+            {
+                var errors = GetDiagnostics(EnvironmentUrlHardcodedRule.Code, text);
+                errors.Should().HaveCount(expectedDiagnosticCount);
+            }
+        }
+
+        [DataRow(0, @"
+param password string
+var sum = 1 + 3
+output sub int = sum
+")]
+        [DataRow(0, @"
+param param1 string
+var location = 'somehost.com'
+output sub int = sum
+")]
+        [DataRow(1, @"
+param param1 string
+var location = 'management.core.windows.net'
```

Comment:
>location [](http://example.com/codeflow?start=4&length=8)

Probably should add some variations in the text - maybe embed the disallowed host deeper in an expression or in various declarations?

Reply:
will add issue for later addition